web_m2x_options/static/src/js/form.js

```diff
@@ -187,7 +187,7 @@ odoo.define('web_m2x_options.web_m2x_options', function (require) {
                     search_more_undef = _.isUndefined(self.options.search_more) && _.isUndefined(self.view.ir_options['web_m2x_options.search_more']),
                     search_more = self.is_option_set(self.view.ir_options['web_m2x_options.search_more']);
 
-                if (values.length > self.limit && (can_search_more || search_more_undef || search_more)) {
+                if ((values.length > self.limit && (can_search_more || search_more_undef || search_more)) || search_more) {
```

Comment:
- First `search_more` is necesary in case is set to False (so the option is disabled).
- Limit = 1 limits also the displayed ocurrences. So this keeps that option independent.